securecookie.go

```diff
@@ -129,17 +129,21 @@ type Codec interface {
 // 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256.
 // The default encoder used for cookie serialization is encoding/gob.
 //
+// decodeKeys are optional, used to authenticate during decoding if the hashKey
+// fails. This allows older hashKeys to be verified during routine key rotation.
+//
 // Note that keys created using GenerateRandomKey() are not automatically
 // persisted. New keys will be created when the application is restarted, and
 // previously issued cookies will not be able to be decoded.
-func New(hashKey, blockKey []byte) *SecureCookie {
+func New(hashKey, blockKey []byte, decodeKeys ...[]byte) *SecureCookie {
 	s := &SecureCookie{
-		hashKey:   hashKey,
-		blockKey:  blockKey,
-		hashFunc:  sha256.New,
-		maxAge:    86400 * 30,
-		maxLength: 4096,
-		sz:        GobEncoder{},
+		hashKey:    hashKey,
+		decodeKeys: decodeKeys,
```

Comment:
If you wanted to do this, you could just add a method that allows adding new (named) keys to a map of keys on the SecureCookie type.

Reply:
I'm entirely on board for instead adding a method to add new named keys to a map, I'll put something together today